Path Canonicalization Failed

Been using this for a couple months & it's been working great ... NO issues! Today, I tried getting this working on a different server & started running into an issue. I've validated the target directory exists BEFORE running this step & it's obviously getting cleaned up but looks like it's having trouble re-creating it via the copy over SSH. 

scp: realpath //app/source: No such file
scp: upload "//app/source": path canonicalization failed
scp: failed to upload directory abcdef to //app/source

Target machine is running Ubuntu 14.04.6 LTS.

**Workflow Usage**
`      - name: Copy via SSH
        uses: sebastianjnuwu/ssh-action-deploy@v3
        with:
          IP: host
          USER: user
          KEY: key
          FOLDER: /app/source
`

**Log**
`Run sebastianjnuwu/ssh-action-deploy@v3
  with:
    IP: XXX.XX.XX.XXX
    USER: ***********
    KEY: ***
    FOLDER: /app/source
    RUN: ls -larth /app;ls -lart /app/source;
    REPO: repo/abcdef
  env:
    key_name: key_abcdef
    dir_target: /app/source
Run mkdir -p ~/.ssh/
  mkdir -p ~/.ssh/
  echo "***
  ***
  ***
  ***
  ***
  ***
  ***" > ~/.ssh/ssh.key
  chmod 600 ~/.ssh/ssh.key
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    key_name: key_abcdef
    dir_target: /app/source
Run cat >>~/.ssh/config <<END
  cat >>~/.ssh/config <<END
  Host ssh
    HostName 172.31.0.10
    User ghdeploy
    Port 22
    IdentityFile ~/.ssh/ssh.key
    StrictHostKeyChecking no
  END
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    key_name: key_abcdef
    dir_target: /app/source
Run ssh ssh rm -rf "/app/source"; 
  ssh ssh rm -rf "/app/source"; 
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    key_name: key_abcdef
    dir_target: /app/source
Run cd ..
  cd ..
  IN="repo/abcdef"
  set -- "$IN" 
  IFS="/"; declare -a Array=($*) 
  scp -r ${Array[1]} ssh://app/source
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    key_name: key_abcdef
    dir_target: /app/source
scp: realpath //app/source: No such file
scp: upload "//app/source": path canonicalization failed
scp: failed to upload directory abcdef to //app/source
Error: Process completed with exit code 1.
`

--- a/action.yml
+++ b/action.yml
@@ -26,12 +26,8 @@ inputs:
     required: false
   COMPATABILITY:
     description: "Add the '-O' compatability flag to the SCP command for targets with older verions of OPENSSH (ie: Ubuntu 14.X)."
-    type: boolean
     required: false
-    default: 'false'
-    options:
-      - true
-      - false
+    type: boolean
 
 runs:
   using: "composite"

--- a/action.yml
+++ b/action.yml
@@ -61,7 +61,7 @@ runs:
         ssh ssh rm -rf "${{ inputs.FOLDER }}"; 
       shell: bash
     - name: 💧 deploy to vps...
-      if: ${{ inputs.COMPATABILITY == false }}
+      if: ${{ !inputs.COMPATABILITY }}
       run: |
        cd ..
        IN="${{ inputs.REPO }}"
@@ -70,7 +70,7 @@ runs:
        scp -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
       shell: bash
     - name: 💧 deploy to vps with compatability...
-      if: ${{ inputs.COMPATABILITY == true }}
+      if: ${{ inputs.COMPATABILITY }}
       run: |
        cd ..
        IN="${{ inputs.REPO }}"

--- a/action.yml
+++ b/action.yml
@@ -32,11 +32,6 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Print Debugging Info
-      shell: bash
-      run: |
-        echo "ENVs:${{ toJSON(env) }}"
-        echo "INPUTs:${{ toJSON(inputs) }}"
     - name: 🗿 Creating ssh files...
       run: |
         mkdir -p ~/.ssh/

--- a/action.yml
+++ b/action.yml
@@ -27,8 +27,11 @@ inputs:
   COMPATABILITY:
     description: "Add the '-O' compatability flag to the SCP command for targets with older verions of OPENSSH (ie: Ubuntu 14.X)."
     type: boolean
-    default: false
     required: false
+    default: 'false'
+    options:
+      - true
+      - false
 
 runs:
   using: "composite"
@@ -61,7 +64,7 @@ runs:
         ssh ssh rm -rf "${{ inputs.FOLDER }}"; 
       shell: bash
     - name: 💧 deploy to vps...
-      if: ${{ !inputs.COMPATABILITY }}
+      if: inputs.COMPATABILITY == 'false'
       run: |
        cd ..
        IN="${{ inputs.REPO }}"
@@ -70,7 +73,7 @@ runs:
        scp -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
       shell: bash
     - name: 💧 deploy to vps with compatability...
-      if: ${{ inputs.COMPATABILITY }}
+      if: inputs.COMPATABILITY == 'true'
       run: |
        cd ..
        IN="${{ inputs.REPO }}"

--- a/action.yml
+++ b/action.yml
@@ -24,6 +24,11 @@ inputs:
   RUN:
     description: "Specify the commands to be executed after the deploy."
     required: false
+  COMPATABILITY:
+    description: "Add the '-O' compatability flag to the SCP command for targets with older verions of OPENSSH (ie: Ubuntu 14.X)."
+    type: boolean
+    default: false
+    required: false
 
 runs:
   using: "composite"
@@ -51,13 +56,23 @@ runs:
         ssh ssh rm -rf "${{ inputs.FOLDER }}"; 
       shell: bash
     - name: 💧 deploy to vps...
+      if: ${{ inputs.COMPATABILITY == false }}
+      run: |
+       cd ..
+       IN="${{ inputs.REPO }}"
+       set -- "$IN" 
+       IFS="/"; declare -a Array=($*) 
+       scp -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
+      shell: bash
+    - name: 💧 deploy to vps with compatability...
+      if: ${{ inputs.COMPATABILITY == true }}
       run: |
        cd ..
        IN="${{ inputs.REPO }}"
        set -- "$IN" 
        IFS="/"; declare -a Array=($*) 
        scp -O -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
-      shell: bash 
+      shell: bash       
     - name: 🍷 Executing commands...
       if: ${{ inputs.RUN }}
       run: ssh ssh "${{ inputs.RUN }}"

--- a/action.yml
+++ b/action.yml
@@ -33,6 +33,12 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Print Debugging Info
+      shell: bash
+      run: |
+        echo "ENVs:${{ toJSON(env) }}"
+        echo "INPUTs:${{ toJSON(inputs) }}"
+        echo "VARs:${{ toJSON(vars) }}"
     - name: 🗿 Creating ssh files...
       run: |
         mkdir -p ~/.ssh/

--- a/action.yml
+++ b/action.yml
@@ -38,7 +38,6 @@ runs:
       run: |
         echo "ENVs:${{ toJSON(env) }}"
         echo "INPUTs:${{ toJSON(inputs) }}"
-        echo "VARs:${{ toJSON(vars) }}"
     - name: 🗿 Creating ssh files...
       run: |
         mkdir -p ~/.ssh/

--- a/action.yml
+++ b/action.yml
@@ -56,7 +56,7 @@ runs:
        IN="${{ inputs.REPO }}"
        set -- "$IN" 
        IFS="/"; declare -a Array=($*) 
-       scp -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
+       scp -O -r ${Array[1]} ssh:/${{ inputs.FOLDER }}
       shell: bash 
     - name: 🍷 Executing commands...
       if: ${{ inputs.RUN }}

